plugins/lookup/merge_variables.py

```diff
@@ -0,0 +1,198 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Thales Netherlands
+# Copyright (c) 2021, Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    author:
+      - Roy Lenferink (@rlenferink)
+      - Mark Ettema (@m-a-r-k-e)
+    name: merge_variables
+    short_description: merge variables with a certain suffix
+    description:
+        - This lookup returns the merged result of all variables in scope that matches the given suffix optionally
+          starting with an initial value.
+    version_added: 6.1.0
+    options:
+      _terms:
+        description:
+          - The suffix of the variable name (all available variables will be checked with endswith(suffix))
+        required: true
+      pattern_type:
+        description:
+          - Change the way of searching for the specified pattern.
+        type: str
+        default: 'regex'
+        choices:
+          - prefix
+          - suffix
+          - regex
+        env:
+          - name: ANSIBLE_MERGE_VARIABLES_PATTERN_TYPE
+      initial_value:
+        description:
+          - An initial value to start with
+        required: false
```

Comment:
`required: false` should be omitted since `false` is the default for `required`. But you should always specify a type. I assume you want to allow any kind of input here?
```suggestion
        type: raw
```

Reply:
Removed the required.

About the type: I actually think that this should be `type: str`, since it allows a prefix, suffix or regex, which are are string values.